cmd/sovereignnode/sovereignNodeRunner.go

```diff
@@ -0,0 +1,1715 @@
+	flagsConfig := configs.FlagsConfig
+	configurationPaths := configs.ConfigurationPathsHolder
+
+	log.Debug("creating healthService")
+	healthService := snr.createHealthService(flagsConfig)
+
+	log.Debug("creating core components")
+	managedCoreComponents, err := snr.CreateManagedCoreComponents(
+		chanStopNodeProcess,
+	)
+	if err != nil {
+		return true, err
+	}
+
+	log.Debug("creating status core components")
+	managedStatusCoreComponents, err := snr.CreateManagedStatusCoreComponents(managedCoreComponents)
+	if err != nil {
+		return true, err
+	}
+
+	log.Debug("creating crypto components")
+	managedCryptoComponents, err := snr.CreateManagedCryptoComponents(managedCoreComponents)
+	if err != nil {
+		return true, err
+	}
+
+	log.Debug("creating network components")
+	managedNetworkComponents, err := snr.CreateManagedNetworkComponents(managedCoreComponents, managedStatusCoreComponents, managedCryptoComponents)
+	if err != nil {
+		return true, err
+	}
+
+	log.Debug("creating disabled API services")
+	webServerHandler, err := snr.createHttpServer(managedStatusCoreComponents)
+	if err != nil {
+		return true, err
+	}
+
+	log.Debug("creating bootstrap components")
+	managedBootstrapComponents, err := snr.CreateManagedBootstrapComponents(managedStatusCoreComponents, managedCoreComponents, managedCryptoComponents, managedNetworkComponents)
+	if err != nil {
+		return true, err
+	}
+
+	snr.logInformation(managedCoreComponents, managedCryptoComponents, managedBootstrapComponents)
+
+	log.Debug("creating data components")
+	managedDataComponents, err := snr.CreateManagedDataComponents(managedStatusCoreComponents, managedCoreComponents, managedBootstrapComponents, managedCryptoComponents)
+	if err != nil {
+		return true, err
+	}
+
+	log.Debug("creating state components")
+	managedStateComponents, err := snr.CreateManagedStateComponents(
+		managedCoreComponents,
+		managedBootstrapComponents,
+		managedDataComponents,
+		managedStatusCoreComponents,
+	)
+	if err != nil {
+		return true, err
+	}
+
+	log.Debug("creating metrics")
+	// this should be called before setting the storer (done in the managedDataComponents creation)
+	err = snr.createMetrics(managedStatusCoreComponents, managedCoreComponents, managedCryptoComponents, managedBootstrapComponents)
+	if err != nil {
+		return true, err
+	}
+
+	log.Debug("registering components in healthService")
+	snr.registerDataComponentsInHealthService(healthService, managedDataComponents)
+
+	nodesShufflerOut, err := bootstrapComp.CreateNodesShuffleOut(
+		managedCoreComponents.GenesisNodesSetup(),
+		configs.GeneralConfig.EpochStartConfig,
+		managedCoreComponents.ChanStopNodeProcess(),
+	)
+	if err != nil {
+		return true, err
+	}
+
+	bootstrapStorer, err := managedDataComponents.StorageService().GetStorer(dataRetriever.BootstrapUnit)
+	if err != nil {
+		return true, err
+	}
+
+	log.Debug("creating nodes coordinator")
+	nodesCoordinatorInstance, err := bootstrapComp.CreateNodesCoordinator(
```

Comment:
> a new instance of the NodesCoordinator with no metachain and just 1 shard should be created here.

I think `CreateNodesCoordinator` takes care of this from the nodes config that it receives.

> But what happens with the metachain processing logic? SystemSCs, ESDT/NFTs, Ratings, Epochs?

I'm not sure how @SebastianMarian implemented/took care of this, honestly. Maybe he can help with some clarification?

Reply:
As there is only one shard, and all the nodes actually are owned / trusted by the business which deployed the solution, there is no need for epochs, shuffle out, rating, jailing, etc. However, we created ValidatorStatsRootHash field into the SovereignChainHeader DTO and in the final of sovereignChainBlockProcessor.ProcessBlock method, we call UpdatePeerState method and we record in each block the resulted validatorStatsRootHash as the value of accountsDB[state.PeerAccountsState].RootHash(). But for now this is used just for tracking and no actions would be taken based on these statistics, as there is no new epoch in sovereign.

Reply:
Related to ESDT, the system vm will be executed to the sovereign shard level, through some vm hooks, but this will be possible after one specific feat which right now is WIP, details could be given by Robert, will be merged into feat/sovereign. Anyway for the first alpha version of sovereign, the ESDT will not be needed / supported.

Reply:
About NodesCoordinator I agree that this component should be refactored for sovereign chain to handle only one shard without metachain. Right now, the modified scripts actually skip the launch of the meta chain nodes from the given .pem file. We should create, based on the same model used for other components, a sovereignNodeCoordinator.go file with the specific sovereign implementation and we should try to use/reuse the common code if it is possible as much as we can.